### PR TITLE
Feat: new resolvers

### PR DESCRIPTION
**Pull request by LucasFalkowsky**
What was done:

- 2 new resolvers were introduced (getAchievementById & getNextStepAchievements)

### CODE REVIEW

File: graphql/user/fields.ts [hunk 1]
```diff
@@ -210,6 +210,18 @@ export class UserFieldsResolver {
         return await getAppointmentsForUser(user, take, skip, cursor, direction);
     }
 
+    @FieldResolver((returns) => [Achievement])
+    @Authorized(Role.ADMIN, Role.OWNER)
+    async achievementById(@Ctx() context: GraphQLContext, @Arg('id') id: number): Promise<Achievement> {
```
Comment (LomyW) [inline graphql/user/fields.ts:215]: can we just call it `achievement`

File: common/achievement/get.ts [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+        include: { template: true },
+    });
+    const achievement = await assembleAchievementData([userAchievement], user);
+    return achievement;
+};
+
+// Next step achievements are sequential achievements that are currently active and not yet completed. They get displayed in the next step card section.
+const getNextStepAchievements = async (user: User): Promise<Achievement[]> => {
+    const userAchievements = await prisma.user_achievement.findMany({
+        where: { userId: user.userID, isSeen: false, template: { type: achievement_type_enum.SEQUENTIAL } },
```
Comment (LomyW) [inline common/achievement/get.ts:21]: do we also have to check here for `achievedAt != null`?

**Review by LomyW — COMMENTED** [review]
looks good to me 👍🏼

File: graphql/user/fields.ts (hunk 1 above)
Comment (LucasFalkowsky) [reply graphql/user/fields.ts:215]: done!

File: common/achievement/get.ts (hunk 2 above)
Comment (LucasFalkowsky) [reply common/achievement/get.ts:21]: No I don't think so. If that was the case we could not get all steps necessary to know the currentStep value. Or the maxSteps value.

File: graphql/user/fields.ts [hunk 3]
```diff
@@ -210,6 +210,18 @@ export class UserFieldsResolver {
         return await getAppointmentsForUser(user, take, skip, cursor, direction);
     }
 
+    @FieldResolver((returns) => [Achievement])
+    @Authorized(Role.ADMIN, Role.OWNER)
+    async achievement(@Ctx() context: GraphQLContext, @Arg('id') id: number): Promise<Achievement> {
```
Comment (dhenkel92) [inline graphql/user/fields.ts:215]:
```suggestion
    async achievements(@Ctx() context: GraphQLContext, @Arg('id') id: number): Promise<Achievement> {
```

File: graphql/user/fields.ts [hunk 4]
```diff
@@ -210,6 +210,18 @@ export class UserFieldsResolver {
...
+        const achievement = await getAchievementById(context.user, id);
+        return achievement;
+    }
+    @FieldResolver((returns) => [Achievement])
+    @Authorized(Role.ADMIN, Role.OWNER)
+    async nextStepAchievements(@Ctx() context: GraphQLContext): Promise<Achievement[]> {
```
Comment (dhenkel92) [inline graphql/user/fields.ts:221]: Maybe rename?
```suggestion
    async achievementsNextSteps(@Ctx() context: GraphQLContext): Promise<Achievement[]> {
```

File: graphql/user/fields.ts (hunk 3 above)
Comment (LucasFalkowsky) [reply graphql/user/fields.ts:215]: no this is achievement. Singular.

File: graphql/user/fields.ts [hunk 5]
```diff
@@ -210,6 +210,18 @@ export class UserFieldsResolver {
         return await getAppointmentsForUser(user, take, skip, cursor, direction);
     }
 
+    @FieldResolver((returns) => [Achievement])
```
Comment (dhenkel92) [inline graphql/user/fields.ts:213]: If it's supposed to return only one achievement, it should be reflected in the annotaion.
```suggestion
    @FieldResolver((returns) => Achievement)
```

File: graphql/user/fields.ts (hunk 4 above)
Comment (LucasFalkowsky) [reply graphql/user/fields.ts:221]: Well this actually is not really the next step of an achievement. The sequential achievements get diesplayed as next step card on the front page.
So if you really would prefer another name we should go for sth like achievementsAsNextSteps. You have any more ideas?

**Pull request merged**